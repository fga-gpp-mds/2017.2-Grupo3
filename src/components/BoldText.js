import React from 'react';
import PropTypes from 'prop-types';
import { View, Text, StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  text: {
    fontSize: 15,
    paddingVertical: 2,
  },
});

const BoldText = ({ data, label }) => (
  <View>
    <Text style={styles.text}>
      <Text style={{ fontWeight: 'bold' }}>
        {label}:
      </Text>
      {data}
    </Text>
  </View>
);

const { string } = PropTypes;

BoldText.propTypes = {
  data: string.isRequired,
  label: string.isRequired,
};

export default BoldText;
