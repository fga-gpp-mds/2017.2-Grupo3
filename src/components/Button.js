import React from 'react';
import PropTypes from 'prop-types';
import { TouchableOpacity, Text, View } from 'react-native';
import styles from '../Styles/GeneralStyles';


class Button extends React.Component {
  defineStyle() {
    if (!this.props.style) {
      if (this.props.enabled) {
        return (
          styles.bigButton
        );
      }
      return (
        styles.disabledBigButton
      );
    }
    return (
      this.props.style.design
    );
  }

  defineTextStyle() {
    if (!this.props.style) {
      return styles.buttonText;
    }
    return this.props.style.text;
  }

  render() {
    return (
      <View>
        <TouchableOpacity
          key={this.props.key}
          style={this.defineStyle()}
          disabled={!this.props.enabled}
          onPress={() => this.props.onPress()}
        >
          <Text style={this.defineTextStyle()}>{this.props.text}</Text>
        </TouchableOpacity>
      </View>
    );
  }
}

const { string, bool, func, element } = PropTypes;

Button.propTypes = {
  key: string.isRequired,
  text: string.isRequired,
  enabled: bool.isRequired,
  onPress: func.isRequired,
  style: element,
};

Button.defaultProps = {
  style: undefined,
};

export default Button;
