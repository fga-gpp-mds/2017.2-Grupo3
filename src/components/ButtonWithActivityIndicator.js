import React from 'react';
import PropTypes from 'prop-types';
import {
  Text, View, TouchableOpacity, ActivityIndicator,
} from 'react-native';
import styles from '../Styles/GeneralStyles';

const ButtonWithActivityIndicator = props => (
  <View>
    {props.isLoading ? (
      <ActivityIndicator
        style={props.activityIndicatorStyle}
        size="large"
        color="#FF9500"
      />
    )
      : (
        <TouchableOpacity
          onPress={() => props.onPress()}
          style={props.buttonStyle}
          activeOpacity={0.7}
          key={props.buttonKey}
        >
          <Text style={styles.buttonText}>{props.buttonText}</Text>
        </TouchableOpacity>
      )}
  </View>
);


// For some reason, the style elements are considered numbers.
ButtonWithActivityIndicator.propTypes = {
  activityIndicatorStyle: PropTypes.number.isRequired,
  isLoading: PropTypes.bool.isRequired,
  onPress: PropTypes.func.isRequired,
  buttonKey: PropTypes.string.isRequired,
  buttonText: PropTypes.string.isRequired,
  buttonStyle: PropTypes.number.isRequired,
};

export default ButtonWithActivityIndicator;
