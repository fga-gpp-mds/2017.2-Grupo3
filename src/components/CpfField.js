import React from 'react';
import PropTypes from 'prop-types';
import { View, TextInput, Text } from 'react-native';
import FontAwesome from 'react-native-vector-icons/FontAwesome';
import styles from '../Styles/GeneralStyles';

const validateCpf = require('validar-cpf');

export default class CpfField extends React.Component {
  state = {
    isValid: false,
    focused: false,
    errorMessage: '',
  };

  setStyle() {
    const style = [styles.InputFieldStyle];
    if (this.state.focused) {
      if (this.state.isValid) {
        style.push(styles.inputFieldValidStyle);
      } else {
        style.push(styles.inputFieldErrorStyle);
      }
    }
    return style;
  }

  validateCpf = (cpf, callback) => {
    const validCpf = cpf.replace(/[^0-9]/g, '');
    const valid = validateCpf(validCpf);
    if (this.state.isValid !== valid) {
      this.setState({ isValid: valid });
    }

    if (!valid) {
      this.state.errorMessage = 'CPF inv√°lido.\n';
    }
    callback(validCpf, valid);
  };
  render() {
    const { value, callback } = this.props;
    return (
      <View>
        <View style={this.setStyle()} >
          <FontAwesome name="user-circle" style={styles.icon} size={26} color="black" />
          <TextInput
            style={styles.InputStyle}
            placeholder="Digite o seu CPF"
            placeholderTextColor="#95a5a6"
            underlineColorAndroid="transparent"
            returnKeyLabel={'next'}
            maxLength={11}
            keyboardType={'numeric'}
            onChangeText={cpf => this.validateCpf(cpf, this.props.callback)}
            value={value}
            onFocus={() => this.setState({ focused: true })}
          />
        </View>

        <View style={{ opacity: !this.state.isValid ? 100 : 0 }}>
          <Text>{this.state.errorMessage}</Text>
        </View>
      </View>
    );
  }
}
CpfField.propTypes = {
  value: PropTypes.string.isRequired,
  callback: PropTypes.func.isRequired,
};
