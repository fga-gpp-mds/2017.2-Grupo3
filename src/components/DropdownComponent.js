import React from 'react';
import PropTypes from 'prop-types';
import { View, Picker } from 'react-native';
import styles from '../Styles/GeneralStyles';
import ErrorMessage from './ErrorMessage';

export default class DropdownComponent extends React.Component {
  state = {
    isValid: false,
    focused: false,
    errorMessage: '',
  };

  setStyle() {
    const style = [styles.InputFieldDropdown];
    if (this.state.focused) {
      if (this.state.isValid) {
        style.push(styles.inputFieldValidStyle);
      } else {
        style.push(styles.inputFieldErrorStyle);
      }
    }
    return style;
  }

  validateDropdown = (checkedAttribute, fieldName, callback) => {
    const valid = (checkedAttribute !== '');

    if (this.state.isValid !== valid) {
      this.setState({ isValid: valid });
    }

    if (!valid) {
      this.state.errorMessage = 'Por favor, selecione um ';
      this.state.errorMessage += fieldName;
    }
    callback(checkedAttribute, valid);
  };

  render() {
    const {
      selectedValue, pickerBody, pickerTitle, fieldName, callback,
    } = this.props;

    return (
      <View>
        <View style={this.setStyle()}>
          <Picker
            onValueChange={checkedAttribute => this.validateDropdown(checkedAttribute, fieldName, callback)}
            selectedValue={selectedValue}
          >
            {pickerTitle}
            {pickerBody}
          </Picker>
        </View>

        <ErrorMessage valid={this.state.isValid} errorText={this.state.errorMessage} />
      </View>
    );
  }
}

DropdownComponent.propTypes = {
  fieldName: PropTypes.string.isRequired,
  selectedValue: PropTypes.string.isRequired,
  callback: PropTypes.func.isRequired,
  pickerTitle: PropTypes.element.isRequired,
  pickerBody: PropTypes.element.isRequired,
};
