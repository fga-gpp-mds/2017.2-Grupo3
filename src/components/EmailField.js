import MaterialIcons from 'react-native-vector-icons/MaterialIcons';
import React from 'react';
import PropTypes from 'prop-types';
import { View, TextInput } from 'react-native';
import styles from '../Styles/GeneralStyles';
import ErrorMessage from './ErrorMessage';

export default class EmailField extends React.Component {
  state = {
    isValid: false,
    focused: false,
    errorMessage: '',
  };

  setStyle() {
    const style = [styles.InputFieldStyle];
    if (this.state.focused) {
      if (this.state.isValid) {
        style.push(styles.inputFieldValidStyle);
      } else {
        style.push(styles.inputFieldErrorStyle);
      }
    }
    return style;
  }

  validateEmail = (email, callback) => {
    const emailRegex = /^[a-zA-Z0-9.!#$%&â€™*+/=?^_`{|}~-]+@[a-zA-Z0-9-]+(?:\.[a-zA-Z0-9-]+)*$/;
    const valid = emailRegex.test(email);
    if (this.state.isValid !== valid) {
      this.setState({ isValid: valid });
    }

    if (!valid) {
      this.state.errorMessage = 'Email deve ser no formato exemplo@exemplo.com\n';
    }
    callback(email, valid);
  };

  render() {
    const {
      size, value, placeholder, onSubmitEditing, callback,
    } = this.props;
    return (
      <View>
        <View style={this.setStyle()}>
          <MaterialIcons name="email" style={styles.icon} size={size} color="black" />
          <TextInput
            width={300}
            style={styles.InputStyle}
            placeholder={placeholder}
            placeholderTextColor="#95a5a6"
            underlineColorAndroid="transparent"
            returnKeyLabel="next"
            maxLength={50}
            keyboardType="email-address"
            autoCapitalize="none"
            onChangeText={email => this.validateEmail(email, callback)}
            value={value}
            onSubmitEditing={() => onSubmitEditing()}
            onFocus={() => this.setState({ focused: true })}
          />
        </View>

        <ErrorMessage valid={this.state.isValid} errorText={this.state.errorMessage} />
      </View>
    );
  }
}


EmailField.propTypes = {
  value: PropTypes.string.isRequired,
  callback: PropTypes.func.isRequired,
  placeholder: PropTypes.string.isRequired,
  onSubmitEditing: PropTypes.func,
  size: PropTypes.number.isRequired,
};

EmailField.defaultProps = {
  width: null,
  onSubmitEditing: () => (() => ({})),
};
