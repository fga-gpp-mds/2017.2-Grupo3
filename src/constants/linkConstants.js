// Link (GET) to authenticate user at Nuvem Cívica and get his Token.
export const AUTHENTICATE_LINK_NUVEM_CIVICA = 'http://mobile-aceite.tcu.gov.br:80/appCivicoRS/rest/pessoas/autenticar';

// Default Link to manipulate users at Nuvem Cívica.
export const DEFAULT_USER_LINK_NUVEM_CIVICA = 'http://mobile-aceite.tcu.gov.br:80/appCivicoRS/rest/pessoas/';

// Link to access posts at Nuvem Cívica.
export const POSTS_LINK_NUVEM_CIVICA = 'http://mobile-aceite.tcu.gov.br:80/appCivicoRS/rest/postagens';

// Link to get schools at Nuvem Cívica.
export const SCHOOL_ENDPOINT = 'http://mobile-aceite.tcu.gov.br:80/nossaEscolaRS/rest/escolas';

// Default Link to manipulate a group at Nuvem Cívica.
export const DEFAULT_GROUP_LINK_NUVEM_CIVICA = 'http://mobile-aceite.tcu.gov.br:80/appCivicoRS/rest/grupos/';

// Link to get posts with full info at Nuvem Cívica.
export const FULL_INFO_POSTS_LINK_NUVEM_CIVICA = 'http://mobile-aceite.tcu.gov.br:80/appCivicoRS/rest/postagens/timeline';
