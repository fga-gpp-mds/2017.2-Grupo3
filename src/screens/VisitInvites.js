import React from 'react';
import PropTypes from 'prop-types';
import axios from 'axios';
import { View, ActivityIndicator, ScrollView, BackHandler } from 'react-native';
import PopupDialog, { DialogTitle, DialogButton } from 'react-native-popup-dialog';
import VisitCard from '../components/VisitCard';
import Header from '../components/Header';
import VisitInfoPopUp from '../components/VisitInfoPopUp';
import { UNABLE_TO_FIND_SCHOOL } from '../constants/generalConstants';
import { SCHOOL_ENDPOINT } from '../constants/linkConstants';
import { logInfo, logWarn } from '../../logConfig/loggers';
import ShowToast from '../components/Toast';
import { backHandlerPop } from '../NavigationFunctions';
import styles from '../Styles/VisitInvistesStyles'

const FILE_NAME = 'VisitInvites';

class VisitInvites extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      visit: {
        content: {
          agentEmail: '',
          codSchool: 0,
          date: '',
          invitedAgent: false,
          visitListOfInvitees: {},
          schoolName: '',
          time: '',
        },
        visitLat: 0,
        visitLong: 0,
      },
    };
  }

  /* When the screen is Mounting, asynchronously calls for
    the list of all visits for which our logged counselor was
    invited.
  */
  componentWillMount() {
    BackHandler.addEventListener('hardwareBackPress', backHandlerPop);

    this.props.asyncGetSchedule(this.props.counselor);
  }

  componentWillUnmount() {
    BackHandler.removeEventListener('hardwareBackPress', backHandlerPop);
  }


  async getLocalization() {
    try {
      const response = await
      axios.get(`${SCHOOL_ENDPOINT}/${this.state.visit.content.codSchool}`, {
        params: {
          campos: 'latitude,longitude',
        },
      });

      logInfo(FILE_NAME, 'getSchoolLocalization in visits Notifications',
        `Successfully got school data: ${JSON.stringify(response.data, null, 2)}`);

      // Since we get the response in portuguese, we need to "translate" it so we
      // can add it in the store.

      await this.setState({
        visit: {
          ...this.state.visit,
          visitLat: response.data.latitude,
          visitLong: response.data.longitude,
        },
      });
    } catch (error) {
      logWarn(FILE_NAME, 'getSchoolLocalization in visits Notifications', error);
      ShowToast.Toast(UNABLE_TO_FIND_SCHOOL);
    }
  }

  render() {
    let activityIndicatorOrCard = null;

    if (this.props.application === true) {
      activityIndicatorOrCard = (
        <ActivityIndicator
          style={{ marginTop: 50, justifyContent: 'center' }}
          size="large"
          color="#FF9500"
        />
      );
    } else {
      activityIndicatorOrCard = this.props.listOfPendingInvitedScheduleList.map(visit => (
        <VisitCard
          visit={visit}
          counselor={this.props.counselor}
          popUpCallBack={async (selectedVisit) => {
            await this.setState({ visit: selectedVisit });
            await this.getLocalization();
            this.popupDialog.show();
          }}
          key={visit.codConteudoPost}
          asyncUpdateSchedule={this.props.asyncUpdateSchedule}
          asyncGetCurrentSchedule={this.props.asyncGetCurrentSchedule}
          currentSchedule={this.props.currentSchedule}
          asyncGetSchedule={this.props.asyncGetSchedule}
        />
      ));
    }

    return (
      <View style={styles.principal}>
        <Header
          title={'Notificações'}
          subTitle={'Visita'}
        />
        <PopupDialog
          /* Popup para mostrar as informações da visita */
          ref={(popupDialog) => {
            this.popupDialog = popupDialog;
          }}
          dialogTitle={<DialogTitle title="Informações sobre a visita" />}
          overlayPointerEvents="none"
          height="60%"
          width="85%"
          actions={[
            <View style={styles.footerPopUp}>
              <DialogButton
                buttonStyle={styles.dialogButtonStyle}
                text="OK"
                onPress={() => this.popupDialog.dismiss()}
                key="dialogButton1"
              />
            </View>,
          ]}
        >
          <VisitInfoPopUp visit={this.state.visit} />
        </PopupDialog>
        <ScrollView style={styles.content}>
          {activityIndicatorOrCard}
        </ScrollView>
      </View>
    );
  }
}

VisitInvites.propTypes = {
  asyncGetSchedule: PropTypes.func.isRequired,
  asyncUpdateSchedule: PropTypes.func.isRequired,
  asyncGetCurrentSchedule: PropTypes.func.isRequired,
  currentSchedule: PropTypes.shape({
    codConteudoPost: PropTypes.number.isRequired,
    codPostagem: PropTypes.number.isRequired,
    content: {
      agentEmail: PropTypes.string.isRequired,
      codSchool: PropTypes.number.isRequired,
      date: PropTypes.string.isRequired,
      invitedAgent: PropTypes.bool.isRequired,
      schoolName: PropTypes.string.isRequired,
      time: PropTypes.string.isRequired,
      visitListOfInvitees: PropTypes.shape({}),
    },
  }),
  counselor: PropTypes.shape({
    name: PropTypes.string.isRequired,
    email: PropTypes.string.isRequired,
    token: PropTypes.string.isRequired,
    nuvemCode: PropTypes.number.isRequired,
  }).isRequired,
  application: PropTypes.bool.isRequired,
  listOfPendingInvitedScheduleList: PropTypes.shape([]).isRequired,
};

VisitInvites.defaultProps = {
  currentSchedule: PropTypes.shape({
    codConteudoPost: 0,
    codPostagem: 0,
    content: {
      agentEmail: 'email@email.com',
      codSchool: 0,
      date: '0-0-0000',
      invitedAgent: false,
      schoolName: 'Nome da escola',
      time: '00:00',
      visitListOfInvitees: PropTypes.shape({}),
    },
  }),
};

export default VisitInvites;
