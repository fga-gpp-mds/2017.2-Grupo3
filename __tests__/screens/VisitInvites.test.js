import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import axios from 'axios';
import thunk from 'redux-thunk';
import MockAdapter from 'axios-mock-adapter';
import Adapter from 'enzyme-adapter-react-16';
import configureStore from 'redux-mock-store';
import VisitInvitesContainer from '../../src/Containers/VisitInvitesContainer';
import VisitInvites from '../../src/screens/VisitInvites';
import { SCHOOL_ENDPOINT } from '../../src/constants/linkConstants';
import * as Toast from '../../src/components/Toast';

const middlewares = [thunk];

Enzyme.configure({ adapter: new Adapter() });

const mockStore = configureStore(middlewares);

const initialState = {
  currentSchedule: {
    codConteudoPost: 0,
    codPostagem: 10,
    content: {
      agentEmail: 'email@email.com',
      codSchool: 0,
      date: '0-0-0000',
      invitedAgent: false,
      schoolName: 'Nome da escola',
      time: '00:00',
      visitListOfInvitees: {},
    },
  },
  counselor: {
    name: 'testCounselor',
    email: 'test@test.com',
    token: 'abc',
    nuvemCode: 123,
  },
  application: {
    isLoading: false,
  },
  list: {
    listOfPendingInvitedScheduleList: [{
      codSchool: 53007654,
      schoolName: 'EC 02 DE CEILANDIA',
      date: '30-12-2017',
      time: '20:36',
      invitedAgent: false,
      agentEmail: '',
      visitListOfInvitees: {
        123: {
          nuvemCode: 123,
          confirmed: false,
          realizedVisit: false,
        },
        6407: {
          nuvemCode: 6407,
          confirmed: true,
          realizedVisit: false,
        },
      },
    }],
  },
  scheduleVisit: {
    currentVisit: {},
  },

  asyncGetSchedule: () => ({}),
};


const store = mockStore(initialState);

describe('Testing VisitInvites', () => {
  it('it renders correctly', () => {
    const wrapper = shallow(
      <VisitInvitesContainer />,
      { context: { store } },
    ).dive();

    expect(wrapper).toMatchSnapshot();
  });

  it('it renders correctly with application loading', () => {
    const newInitialState = initialState;
    newInitialState.application.isLoading = true;
    const newStore = mockStore(newInitialState);
    const wrapper = shallow(
      <VisitInvitesContainer />,
      { context: { store: { ...newStore } } },
    ).dive();

    expect(wrapper).toMatchSnapshot();
  });
});

describe('Testing VisitInvites prototype methods', () => {
  const wrapper = shallow(<VisitInvitesContainer />, { context: { store } }).dive();
  const state = initialState.currentSchedule.content;

  wrapper.instance().setState({
    visit: {
      content: {
        ...state,
      },
    },
  });

  it('Testing getLocalization on sucess', async () => {
    const spy = jest.spyOn(VisitInvites.prototype, 'getLocalization');

    const mock = new MockAdapter(axios);

    mock.onGet(`${SCHOOL_ENDPOINT}/${initialState.currentSchedule.content.codSchool}`, {
      params: {
        campos: 'latitude,longitude',
      },
    }).reply(200, { latitude: -200, longitude: -300 });

    await wrapper.instance().getLocalization();
    expect(spy).toHaveBeenCalled();
    spy.mockClear();
  });

  it('Testing getLocalization on failure', async () => {
    const spy = jest.spyOn(VisitInvites.prototype, 'getLocalization');

    const mock = new MockAdapter(axios);

    mock.onGet(`${SCHOOL_ENDPOINT}/${initialState.currentSchedule.content.codSchool}`, {
      params: {
        campos: 'latitude,longitude',
      },
    }).reply(400);

    Toast.default.Toast = jest.fn(param => ({}));

    try {
      await wrapper.instance().getLocalization();
    } catch (error) {
      expect(Toast.default.Toast.mock.calls.length).toBe(1);
    }
    spy.mockClear();
  });
});
