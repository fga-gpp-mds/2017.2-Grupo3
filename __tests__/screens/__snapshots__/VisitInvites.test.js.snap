// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing VisitInvites it renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Header
    backButton={false}
    onPress={[Function]}
    subTitle="Visita"
    title="Notificações"
  />
  <PopupDialog
    actions={
      Array [
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#F9F9FB",
              "borderBottomLeftRadius": 8,
              "borderBottomRightRadius": 8,
              "borderColor": "#DAD9DC",
              "borderTopWidth": 0.5,
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <DialogButton
            align="center"
            buttonStyle={
              Object {
                "marginVertical": -10,
              }
            }
            disabled={false}
            onPress={[Function]}
            text="OK"
          />
        </View>,
      ]
    }
    dialogTitle={
      <DialogTitle
        haveTitleBar={true}
        title="Informações sobre a visita"
        titleAlign="center"
      />
    }
    height="60%"
    overlayPointerEvents="none"
    width="85%"
  >
    <VisitInfoPopUp
      visit={
        Object {
          "content": Object {
            "agentEmail": "",
            "codSchool": 0,
            "date": "",
            "invitedAgent": false,
            "schoolName": "",
            "time": "",
            "visitListOfInvitees": Object {},
          },
          "visitLat": 0,
          "visitLong": 0,
        }
      }
    />
  </PopupDialog>
  <ScrollViewMock
    style={
      Object {
        "backgroundColor": "white",
        "flex": 1,
        "paddingTop": 6,
      }
    }
  >
    <VisitCard
      asyncGetCurrentSchedule={[Function]}
      asyncGetSchedule={[Function]}
      asyncUpdateSchedule={[Function]}
      counselor={
        Object {
          "email": "test@test.com",
          "name": "testCounselor",
          "nuvemCode": 123,
          "token": "abc",
        }
      }
      currentSchedule={Object {}}
      popUpCallBack={[Function]}
      visit={
        Object {
          "agentEmail": "",
          "codSchool": 53007654,
          "date": "30-12-2017",
          "invitedAgent": false,
          "schoolName": "EC 02 DE CEILANDIA",
          "time": "20:36",
          "visitListOfInvitees": Object {
            "123": Object {
              "confirmed": false,
              "nuvemCode": 123,
              "realizedVisit": false,
            },
            "6407": Object {
              "confirmed": true,
              "nuvemCode": 6407,
              "realizedVisit": false,
            },
          },
        }
      }
    />
  </ScrollViewMock>
</View>
`;

exports[`Testing VisitInvites it renders correctly with application loading 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Header
    backButton={false}
    onPress={[Function]}
    subTitle="Visita"
    title="Notificações"
  />
  <PopupDialog
    actions={
      Array [
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#F9F9FB",
              "borderBottomLeftRadius": 8,
              "borderBottomRightRadius": 8,
              "borderColor": "#DAD9DC",
              "borderTopWidth": 0.5,
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <DialogButton
            align="center"
            buttonStyle={
              Object {
                "marginVertical": -10,
              }
            }
            disabled={false}
            onPress={[Function]}
            text="OK"
          />
        </View>,
      ]
    }
    dialogTitle={
      <DialogTitle
        haveTitleBar={true}
        title="Informações sobre a visita"
        titleAlign="center"
      />
    }
    height="60%"
    overlayPointerEvents="none"
    width="85%"
  >
    <VisitInfoPopUp
      visit={
        Object {
          "content": Object {
            "agentEmail": "",
            "codSchool": 0,
            "date": "",
            "invitedAgent": false,
            "schoolName": "",
            "time": "",
            "visitListOfInvitees": Object {},
          },
          "visitLat": 0,
          "visitLong": 0,
        }
      }
    />
  </PopupDialog>
  <ScrollViewMock
    style={
      Object {
        "backgroundColor": "white",
        "flex": 1,
        "paddingTop": 6,
      }
    }
  >
    <ActivityIndicator
      animating={true}
      color="#FF9500"
      hidesWhenStopped={true}
      size="large"
      style={
        Object {
          "justifyContent": "center",
          "marginTop": 50,
        }
      }
    />
  </ScrollViewMock>
</View>
`;
